File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = [[]]
        n = 1<<len(nums)
        for i in range(1,n):
            res = []
            for j in range (len(nums)):
                if(i & ( 1<<j)):
                    res.append(nums[j])
            ans.append(res)
        return ans

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>>res;
    void go(vector<int>&nums,vector<int>temp,int idx){
        if(idx==nums.size()){
            res.push_back(temp);
            return ;
        }
        go(nums,temp,idx+1);
        temp.push_back(nums[idx]);
        go(nums,temp,idx+1);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        res.clear();
        vector<int>temp;
        go(nums,temp,0);
        return res;
    }
};